src/dotnet/ToolManifest/ToolManifestEditor.cs

```diff
@@ -63,6 +68,11 @@ public ToolManifestEditor(IFileSystem fileSystem = null, IDangerousFileDetector
                     nuGetVersion.ToNormalizedString()));
             }
 
+            if (deserializedManifest.Tools == null)
+            {
+                deserializedManifest.Tools = new Dictionary<string, SerializableLocalToolSinglePackage>();
+            }
+
             deserializedManifest.Tools.Add(
                 packageId.ToString(),
```

Comment:
Unrelated to this change: do we need to normalize the package ids at all (I forget what `ToString`'s default implementation is for `PackageId`)?  Given a few historical issue we have had regarding the casing of package ids for global tools, I just want to check.

Reply:
it does. All strong type here is normalized with proper equality